pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -770,3 +770,215 @@ func TestNewRSNewReplicas(t *testing.T) {
 		}
 	}
 }
+
+func TestIsDeploymentComplete(t *testing.T) {
```

Comment:
Add unit tests for the condition removal / checking stuff (code like that always tends to be slightly fragile).  It's fine if you test one of the larger methods with it in the controller, but I want a test somewhere that verifies that conditions are correctly and safely added, removed, updated, etc.